### PR TITLE
[Collections] Adds ability to programmatically update the position of the reordering cell

### LINKED ISSUE #5559: [Collections] Allow MDCCollectionViewController to update the position of the reordering cell programatically
## Feature Request/Suggestion

The MDCollectionViewEditor correctly updates the moving cell's location but ONLY when the user changes their touch location. In the case where the moving cell's position is updated due to a collection view update, the moving cell's location does not stay at the user's touch location before the update.

It would be nice if a MDCCollectionViewController subclass could programmatically prompt the MDCCollectionViewEditor to update the moving cell's position to the user's touch location when it knows it has issued an update in the collection view which may change its position.

## Additional context

If you are a Googler, see b/118684230 for reproduction of this behavior.

---
#### Internal data
- Associated internal bug: [b/118761267](http://b/118761267)

### PR DESCRIPTION
**Pull request by felixaemiliano**
Closes #5559

### CODE REVIEW

File: components/Collections/src/MDCCollectionViewEditing.h [hunk 1]
```diff
@@ -55,4 +55,15 @@
...
+/**
+ Updates the position of the cell being reordered to match the user's touch location. This method
+ is a no-op if a cell is not currently being reordered.
+ 
+ While the reordering cell's location is updated automatically to match the user's touch location,
+ this method allows for the explicit update of the cell's position which may be required when the
+ cell's location changes due to a layout update (such as performing collection view changes while
+ the user is reordering).
+ */
+- (void)updateReorderCellPosition;
```
Comment (jverkoey) [inline components/Collections/src/MDCCollectionViewEditing.h]: Please also answer the following questions in the docs:

- When is this API typically meant to used?
- From where is this API typically meant to be used?

E.g. "Call foo from your Bar's doLayout implementation."

Comment (jverkoey) [inline components/Collections/src/MDCCollectionViewEditing.h:71]: These docs describe when to use this API — can we also provide additional context such as which class or method we might call this API from?

Comment (felixaemiliano) [reply components/Collections/src/MDCCollectionViewEditing.h:71]: Would something like this be what you're asking for:

### Example

- (void)collectionView:(nonnull UICollectionView *)collectionView
    willBeginDraggingItemAtIndexPath:(nonnull NSIndexPath *)indexPath {
    [self.collectionView performBatchUpdates: ^{
      // Perform set up for moving, such as collapsing sections or any other animations pre-move.
  } completion:^(BOOL finished) {
      // Ensure that the reordering cell is under user's touch after the updates are performed.
      [self.editor updateReorderCellPosition];
  }];
}

Or would it be sufficient to add:

"For example, you can call this in the the MDCCollectionViewEditingDelegate's implementation of -collectionView:willBeginDraggingItemAtIndexPath:, after performing updates to the collection view in preparation for the user's move of an item."

Comment (jverkoey) [reply components/Collections/src/MDCCollectionViewEditing.h:71]: > "For example, you can call this in the the MDCCollectionViewEditingDelegate's implementation of -collectionView:willBeginDraggingItemAtIndexPath:, after performing updates to the collection view in preparation for the user's move of an item."

This is super helpful!

**Review by jverkoey — APPROVED** [review]
LGTM pending documentation additions.

**Pull request merged**